packages/fxa-settings/src/models/experiments/key-stretch-experiment.ts

```diff
@@ -21,11 +23,69 @@ export class KeyStretchExperiment extends ModelDataProvider {
   @bind()
   forceExperimentGroup?: string;
 
-  isV2() {
-    return (
-      this.stretch === '2' ||
-      (this.forceExperiment === 'key-stretch' &&
-        this.forceExperimentGroup === 'v2')
+  isV2(config: Pick<Config, 'featureFlags'>) {
+    // If stretch=2 in URL, then force V2 key stretching for this page render,
+    // regardless of all other settings / configs. This is used mostly for
```

Comment:
Don't know if there are any risks&hellip; but if this _is_ just for dev/test purposes, should we add any checks to make sure we aren't running this code path on production env (unless that is overly complex due to Docker-ing)?

Reply:
Yep, it's for dev/test only. I gennerally avoid conditional logic against the env as this is much better accomplished with config.

I personally don't see this being a risk since there are other query parameters that can turn this on. One thing to keep in mind, is that we might want to turn it on for a single step, but not the entire flow. That's another reason to have this override.

I can see an argument for putting this if statement after the feature flag check though... so I'll do that since it caught some scrutiny.

Reply:
Yeah, my thinking was we would want to be able to disable it entirely with the feature flag (SRE being able to do this without a retag is super nice), but in that case I don't think we even need the param since we could use the force experiment params. 

I'm unsure of test cases we could benefit from by letting it override the feature flag, but also not sure what real security concerns there could be (what's the worst case scenario... some small sector of users might try messing with it and might get their account in a weird state?) so I'll leave it up to you to decide.

Reply:
I don't think there is much risk. I think the real use case is that we want to test a single step of the login flow with quick stretch enabled, but not the rest. If we use the experiment parameters, then we end up enrolling and having to go through content server. This query parameter flag was definitely useful for testing isolated behavior when V2 key stretching is turned on. 

I will definitely give the feature flag precedence though!